apis/v1alpha1/backendpolicy_types.go

```diff
@@ -0,0 +1,232 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// BackendPolicy defines policies associated with backends such as Services.
+type BackendPolicy struct {
+	metav1.TypeMeta   `json:",inline" protobuf:"bytes,1,opt,name=typeMeta"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,2,opt,name=metadata"`
+
+	Spec   BackendPolicySpec   `json:"spec,omitempty" protobuf:"bytes,3,opt,name=spec"`
+	Status BackendPolicyStatus `json:"status,omitempty" protobuf:"bytes,4,opt,name=status"`
+}
+
+// +kubebuilder:object:root=true
+
+// BackendPolicyList contains a list of BackendPolicy
+type BackendPolicyList struct {
+	metav1.TypeMeta `json:",inline" protobuf:"bytes,1,opt,name=typeMeta"`
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,2,opt,name=metadata"`
+	Items           []BackendPolicy `json:"items" protobuf:"bytes,3,rep,name=items"`
+}
+
+// BackendPolicySpec defines desired policy for a Backend.
+type BackendPolicySpec struct {
+	// BackendRefs define which backends this policy should be applied to. If
+	// more than one BackendPolicy targets the same Backend, precedence will be
+	// given to the oldest BackendPolicy.
+	//
+	// Support: Core
+	//
+	// +optional
+	BackendRefs []BackendRef `json:"backendRefs" protobuf:"bytes,1,opt,name=backendRefs"`
```

Comment:
I am not sure a list of references is a very common case. I guess it doesn't hurt to have a list here vs copying the spec over. But a "global defaults" is a common request we get for these type of options

Reply:
Now that the scope of this PR has been reduced to TLS config a global default might not be as helpful. Definitely agree that it could be helpful to have global defaults for some potential future additions to BackendPolicy. Defining these defaults could also potentially be accomplished with defaults for GatewayClass params. 

I agree that multiple BackendRefs are not a very common use case, but it seemed simple enough to support and added some extra flexibility for users.